pkg/controller/deployment/deployment_controller.go

```diff
@@ -341,6 +397,11 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 
+	err = dc.classifyReplicaSets(deployment)
```

Comment:
I think classifying replica sets should happen after we handle overlapping deployments.